src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -60,10 +60,12 @@ namespace ErrorCodes
     extern const int BAD_ARGUMENTS;
 }
 
-AsynchronousInsertQueue::InsertQuery::InsertQuery(const ASTPtr & query_, const Settings & settings_)
+AsynchronousInsertQueue::InsertQuery::InsertQuery(const ASTPtr & query_, const Settings & settings_, const std::optional<UUID> & user_id_, const std::vector<UUID> & current_roles_)
     : query(query_->clone())
     , query_str(queryToString(query))
     , settings(settings_)
+    , user_id(user_id_)
+    , current_roles(current_roles_)
```

Comment:
Those fields should be added to hash and `==` operator.

Reply:
done